v2/pkg/core/execute_options.go

```diff
@@ -60,13 +60,16 @@ func (e *Engine) ExecuteScanWithOpts(templatesList []*templates.Template, target
 	// Execute All SelfContained in parallel
 	e.executeAllSelfContained(selfContained, results, selfcontainedWg)
 
+	var strategyResult *atomic.Bool
 	switch e.options.ScanStrategy {
 	case "template-spray":
-		results = e.executeTemplateSpray(filtered, target)
+		strategyResult = e.executeTemplateSpray(filtered, target)
 	case "host-spray":
-		results = e.executeHostSpray(filtered, target)
+		strategyResult = e.executeHostSpray(filtered, target)
 	}
 
+	results.CompareAndSwap(false, strategyResult.Load())
```

Comment:
@Mzack9999 what do you think of adding `default` case to switch case . if someone is using nuclei sdk and scanstrategy is not one them it will panic
due to `strategyResult.Load()`